Processing_data/Classes/StatsBucket.cs

```diff
@@ -0,0 +1,41 @@
+using Processing_data.Interfaces;
+
+namespace Processing_data.Classes
+{
+    /// <summary>
+    /// Represents a collection of statistics for a set of data
+    /// </summary>
+    public class StatsBucket : Dictionary<Coordinate, Statistics>
+    {
+        private IPrint _printingService;
+        /// <summary>
+        /// Adds a data value to the statistics bucket
+        /// </summary>
+        /// <param name="coordinate"></param>
+        /// <param name="data"></param>
+        public void AddData(Coordinate coordinate, double data)
+        {
+            if (!ContainsKey(coordinate))
+            {
+                Add(coordinate, new Statistics(new List<double> { data }));
+            }
+            else
+            {
+                this[coordinate].AddData(data);
+            }
+        }
+
+        public StatsBucket(IPrint printingService) : base(new CustomEqualityComparer())
```

Comment:
IMO adding a dependency on a presentation service here is a bad idea. A bucket should know how to store Statistics (SRP). Moreover, IPrint already depends on the StatsBucket.